.github/workflows/ci.yml

```diff
@@ -19,7 +19,9 @@ jobs:
         sudo ./ci/setup_cmake.sh
         sudo ./ci/setup_ci_environment.sh
     - name: run cmake tests (without otlp-exporter)
-      run: ./ci/do_ci.sh cmake.test
+      run: |
+        sudo ./ci/setup_thrift.sh
```

Comment:
We don't need `sudo`, as github actions are run by default as `root` user.
Also, if we move thrift installation as part of do_ci.sh script, this script will remain usable outside of github workflow, without separately installing thrift ( similar to how it is done for `prometheus`). Probably `setup_thrift.sh`  can expose a function to install thrift, and that function can be called from do_ci.sh for actual installation if called with "cmake.test" argument.

Reply:
I also believe `sudo` is not necessary, just follow other scripts invocation like `setup_grpc.h`.

I don't prefer moving the installation logic to `do_ci.sh` which would make `do_ci.sh` too complicated with the third-party dependence setup instructions, and also hard to share with non-CI environment. Expose a setup function in `setup_thrift.sh` looks good, but we still need include `setup_thrift.sh` to invoke the function. So it may be simpler to just invoke the script?

Reply:
@lalitb do all github action run as root by default? I just removed all the `sudo` in our `ci.yml` but got permission denied in the CI run.

Reply:
I thought unless explicitly specified with `--user` options, all jobs run as `root` user. Probably revert it back if there is a permission denied error. Sorry for the confusion.